### PR TITLE
add LindenmayerSystemSnippet & tests and Update README.md

### PR DESCRIPTION
**Pull request by SayedQassim**
Issues
- [x] #214

### CODE REVIEW

File: src/test/java/string/LindenmayerSystemSnippetTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    assertEquals(
+        "ABAAB",
+        LindenmayerSystemSnippet.generateLindenmayerSystem(axiom, productionRules, 3)
+    );
+    assertEquals(
+        axiom,
+        LindenmayerSystemSnippet.generateLindenmayerSystem(axiom, productionRules, 0)
+    );
+  }
+}
```
Comment (Ritabrata1080) [inline src/test/java/string/LindenmayerSystemSnippetTest.java:70]: Please add an empty/blank line at EOF to satisfy github.

Comment (SayedQassim) [reply src/test/java/string/LindenmayerSystemSnippetTest.java]: Hi @Ritabrata1080, Done.

**Review by Ritabrata1080 — APPROVED** [review]
Changes LGTM

File: src/test/java/string/LindenmayerSystemSnippetTest.java (hunk 1 above)
Comment (Ritabrata1080) [reply src/test/java/string/LindenmayerSystemSnippetTest.java]: Approved @SayedQassim ✅. 

please consider squashing multiple commits to maintain linear history on master.

Comment (SayedQassim) [reply src/test/java/string/LindenmayerSystemSnippetTest.java]: @Ritabrata1080 Done.

Comment (Ritabrata1080) [reply src/test/java/string/LindenmayerSystemSnippetTest.java]: @iluwatar  plz review this PR and also #226. 

cc : @SayedQassim

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

**Pull request merged**